botocore/exceptions.py

```diff
@@ -395,6 +414,12 @@ def _get_retry_info(self, response):
                                   metadata['RetryAttempts'])
         return retry_info
 
+    def __reduce__(self):
+        # Subclasses of ClientError's are dynamically generated and
+        # cannot be pickled unless they are attributes of a
+        # module. So at the very least return a ClientError back.
+        return ClientError, (self.response, self.operation_name)
```

Comment:
This is a sub class of `BotoCoreError` and not `ClientError`.

Reply:
This ***is*** `ClientError`

Reply:
Oh, for some reason I though this was in `ConnectionError` and was really confused.